mitosheet/mitosheet/api/get_params.py

```diff
@@ -27,6 +28,14 @@ def get_params(params: Dict[str, Any], steps_manager: StepsManagerType) -> Optio
         if step.step_id == step_id_to_match:
             found_params = step.params
         
+        if step.params is not None and params_to_match is not None and len(params_to_match) > 0:
+            all_matched = True
+            for key, value in params_to_match.items():
+                if step.params[key] != value:
```

Comment:
I think we could change this `[key]` access to a .get - both here and above in this function -- just to protect ourselves